### PR TITLE
Fixed how we were str joining the buffer for the proxied request body.

### LINKED ISSUE #272: Extended chars in the status field of devices not handled properly
Trying to post `ℂ◉℗⒴ ℘ⓐṨͲℰ Ⓒℌ◭ℝ◬ℂ⒯℮ℛ ` to a display replaced it with `????????…`

Maybe the Zetta API doesn't seem to support extended characters in the status field of devices?

zetta_demo___nest

### PR DESCRIPTION
**Pull request by AdamMagaluk**
Fixes #272

### CODE REVIEW

File: lib/http_server.js [hunk 1]
```diff
@@ -446,10 +446,34 @@ ZettaHttpServer.prototype.proxyToPeer = function(env, next) {
...
+        while ((chunk = response.read()) != null) {
+          buf.push(chunk);
+          len += chunk.length;
+        }
+
+        if (!buf.length) {
+          return;
+        }
+
+        if (buf.length && Buffer.isBuffer(buf[0])) {
```
Comment (kevinswiber) [inline lib/http_server.js]: When would buf[0] not be a buffer?

File: test/test_api.js [hunk 2]
```diff
@@ -809,6 +809,16 @@ describe('Zetta Api', function() {
         .end(done);
     });
 
+    it('device action should support extended characters', function(done) {
+      request(getHttpServer(app))
+        .post(url)
+        .type('form')
+        .send({ action: 'test-text', value: "🙌💸🙌" })
```
Comment (mdobson) [inline test/test_api.js]: 👑would have also been acceptable

**Comment by kevinswiber** [comment]
Aside from the one line note, looks good to me!

**Comment by mdobson** [comment]
+1

File: lib/http_server.js (hunk 1 above)
Comment (AdamMagaluk) [inline lib/http_server.js]: You would think, I grabbed that piece of code from our get_body.js in zetta-proxy not sure where it originally came from.

Comment (AdamMagaluk) [inline lib/http_server.js]: Which it looks like it came from argo https://github.com/argo/argo/blob/9e39f7089a367f156325c2e727c6d5f89ba1b710/argo.js#L66-L91.

**Comment by AdamMagaluk** [comment]
@kevinswiber Removed that conditional since it wasn't useful.

File: lib/http_server.js (hunk 1 above)
Comment (kevinswiber) [inline lib/http_server.js]: Might be a leftover from upgrading to Streams2.  We should probably kill it everywhere.  Or better yet, replace it with something a little safer.  That's an OOM attack waiting to happen.  Most solutions (Apache, IIS) will begin to buffer to disk until a defined maximum is reached.

**Pull request merged**